src/rgw/rgw_rados.h

```diff
@@ -1313,10 +1313,12 @@ class RGWRados
   RGWDataChangesLog *data_log;
 
   virtual ~RGWRados() {
-    for (uint32_t i=0; i < num_rados_handles; i++) {
-      if (rados[i]) {
-        rados[i]->shutdown();
-        delete rados[i];
+    if (rados){
```

Comment:
Might be, but the idea is to make each 'rados[i]' to NULL (and also 'rados' to be NULL, since we are defensive anyways!) during the failure scenario in init_rados(), such that you don't need to do worry about anything in the destructor.